File: frontend/ATS/src/app/shared/models/stages/stage.ts
import { Action } from '../action/action';
import { Review } from '../reviews/review';

export interface Stage {
  id: string;
  name: string;
  type: number;
  actions: Action[];
  // stageType Type { get; set; }
  dataJson: string | null | undefined;
  index: number;
  IsReviewable: boolean;
  vacancyId: string;
  reviews: Review[];
}

File: backend/src/Domain/Entities/Stage.cs
using Domain.Enums;
using Domain.Common;
using System.Collections.Generic;

namespace Domain.Entities
{
    public class Stage : Entity
    {
        public string Name { get; set; }
        public StageType Type { get; set; }
        public int Index { get; set; }
        public bool IsReviewable { get; set; }
        public string VacancyId { get; set; }
        public string? DataJson { get; set; }

        public Vacancy Vacancy { get; set; }
        public ICollection<Action> Actions { get; set; }
        public ICollection<CandidateToStage> CandidateToStages { get; set; }
        public ICollection<ReviewToStage> ReviewToStages { get; set; }
        public ICollection<CandidateReview> Reviews { get; set; }
        public ICollection<CandidateComment> CandidateComments { get; set; }
    }
}